### DISCUSSION TITLE
Auth UI for Flutter

### DISCUSSION THREAD

**Discussion by rubensdemelo** [supabase/supabase#3612] [Feature Requests]
Firebase and Amplify are working on an auth interface to easily handle user authentication. 

https://github.com/aws-amplify/amplify-flutter/pull/951

https://github.com/FirebaseExtended/flutterfire/tree/%40invertase/ui/packages/firebase_ui

Can we start working on it too ?

**Comment by MildTomato** [comment]
@rubensdemelo I'd be interested in starting a project were we support a Auth UI for react, react-native, vuejs, flutter (and others..).

We currently have a react Auth component in https://github.com/supabase/ui but we could really expand on it, move it to its own repo, and create an interface for each framework that works out of the box.

Hosted auth UI is also another interesting topic of discussion. If anyone is interested in this, perhaps [start a new discussion thread](https://github.com/supabase/supabase/discussions/new?category=ideas).

**Reply by MildTomato** [reply]
Perhaps part of this is also just agreeing on a standard between all the Auth interfaces, so they work as similarly as possible, behave roughly the same and also have the same theming, and ability to apply custom themes.

The closest example to something like this i can think of (other than [firebase ui web](https://github.com/firebase/firebaseui-web)) is probably [Stripe Elements](https://stripe.com/docs/payments/payment-element). Although an Auth interface is probably a bit more straightforward technically, since we won't need to use iframes etc.

**Reply by MildTomato** [reply]
We have actually started this project now.
[Auth UI](https://github.com/supabase-community/auth-ui)
It currently only supports react via npm.

Is anyone interested in created a flutter version of this ?

**Comment by dshukertjr** [comment]
We have a developer preview version of Supabase Auth UI for Flutter now! Stable release coming in a few weeks!
https://pub.dev/packages/supabase_auth_ui
